python/pyarrow/_dataset_parquet.pyx

```diff
@@ -56,9 +63,163 @@ from pyarrow._parquet cimport (
 
 cdef Expression _true = Expression._scalar(True)
 
-
 ctypedef CParquetFileWriter* _CParquetFileWriterPtr
 
+IF PARQUET_ENCRYPTION_ENABLED:
+    cdef class ParquetEncryptionConfig(_Weakrefable):
+        """
+        Configuration for Parquet Encryption.
+
+        Parameters
+        ----------
+        crypto_factory : CryptoFactory
+            Factory for creating cryptographic instances.
+        kms_connection_config : KmsConnectionConfig
+            Configuration for connecting to Key Management Service.
+        encryption_config : EncryptionConfiguration
```

Comment:
```suggestion
        encryption_config : pyarrow.parquet.encryption.EncryptionConfiguration
```

This is long, but I think being explicit with the full name helps to understand which objects are from the parquet module and which objects from the dataset module (as that still confuses me in this PR, especially with the very similar names EncryptionConfiguration vs ParquetEncryptionConfig)

(and the same for the ParquetDecryptionConfig docstring)